### PR TITLE
HFnose: adapting calo towers

### PR DESCRIPTION
**Pull request by mariadalfonso**
here one validation plots with singlePhotons
http://dalfonso.web.cern.ch/dalfonso/HGCAL/L1_gamma_pt20.png

### CODE REVIEW

File: L1Trigger/L1THGCal/interface/HGCalTriggerTowerGeometryHelper.h [hunk 1]
```diff
@@ -38,6 +38,8 @@ class HGCalTriggerTowerGeometryHelper {
   unsigned short getTriggerTower(const l1t::HGCalTriggerCell&) const;
   unsigned short getTriggerTower(const l1t::HGCalTriggerSums&) const;
 
+  bool doNose_;
```
Comment (jbsauvan) [inline L1Trigger/L1THGCal/interface/HGCalTriggerTowerGeometryHelper.h]: Should it be `private`? `doNose_` is meant to be internal, right?

File: DataFormats/L1THGCal/interface/HGCalTowerID.h [hunk 2]
```diff
@@ -25,7 +27,9 @@ namespace l1t {
     unsigned short rawId() const { return rawId_; }
 
   private:
-    unsigned short rawId_;
+    uint32_t rawId_;
+    static const int subDetMask = 0x1;  // two for now 0 is HGC and 1 is HFNose
```
Comment (jbsauvan) [inline DataFormats/L1THGCal/interface/HGCalTowerID.h:31]: An enum could be used to make explicit what 0 and 1 mean

File: L1Trigger/L1THGCal/interface/HGCalTriggerTowerGeometryHelper.h (hunk 1 above)
Comment (mariadalfonso) [reply L1Trigger/L1THGCal/interface/HGCalTriggerTowerGeometryHelper.h]: done in commit 1208015

File: DataFormats/L1THGCal/interface/HGCalTowerID.h (hunk 2 above)
Comment (mariadalfonso) [reply DataFormats/L1THGCal/interface/HGCalTowerID.h:31]: I didn't find a way to use the enum as I do not use 0 and 1 explicitly.
as now I pass a 'short subdetIsNode' so 0 and 1 should be clear

Comment (jbsauvan) [reply DataFormats/L1THGCal/interface/HGCalTowerID.h:31]: Ok, that's fine

**Comment by jbsauvan** [comment]
Could you run `code-format` before I merge

**Comment by mariadalfonso** [comment]
shall  I squash all the commits ?

**Comment by jbsauvan** [comment]
Yes, thanks!

**Comment by mariadalfonso** [comment]
> Yes, thanks!

done

**Pull request merged**